components/sup/src/manager/commands.rs

```diff
@@ -185,81 +184,39 @@ pub fn service_load(mgr: &ManagerState,
+        let force = opts.force.unwrap_or(false);
+        // We've seen this service  before. Thus `load`
+        // basically acts as a way to edit spec files on the
+        // command line. As a result, we a) check that you
+        // *really* meant to change an existing spec, and b) DO
+        // NOT download a potentially new version of the package
+        // in question
+        if !force {
+            return Err(net::err(ErrCode::Conflict,
+                                format!("Service already loaded, unload '{}' \
+                                         and try again",
+                                        ident)));
         }
-        Some(mut spec) => {
-            // We've seen this service  before. Thus `load`
-            // basically acts as a way to edit spec files on the
-            // command line. As a result, we a) check that you
-            // *really* meant to change an existing spec, and b) DO
-            // NOT download a potentially new version of the package
-            // in question
-
-            if !force {
-                return Err(net::err(ErrCode::Conflict,
-                                    format!("Service already loaded, unload '{}' \
-                                             and try again",
-                                            ident)));
-            }
+        spec
+    } else {
+        ServiceSpec::default()
+    };
 
-            opts.into_spec(&mut spec);
-
-            // Only install if we don't have something
-            // locally; otherwise you could potentially
-            // upgrade each time you load.
-            //
-            // Also make sure you're pulling from where you're
-            // supposed to be pulling from!
-
-            // TODO (DM): Why do we use `spec.bldr_url` and `spec.channel` here?
-            // `opts.into_spec` will override these fields if they are defined in the `ServiceLoad`
-            // so we are essentially always using `bldr_url` and `bldr_channel`. In fact `bldr_url`
-            // and `bldr_channel` are not even neccessary, we could just always use what
-            // is in `spec` because `opts.into_spec` will set it appropriatly.
-            // Potentially, I could see the reason for the use of `spec` here trying to
-            // prevent updating the `bldr_url` and `channel` on a --force reload, but that
-            // is not the current behavior. With this change this method could be greatly
-            // simplified.
-            let package =
-                util::pkg::satisfy_or_install(req, &source, &spec.bldr_url, &spec.channel)?;
-            if let Err(e) = spec.validate(&package) {
-                return Err(net::err(ErrCode::InvalidPayload,
-                                    format!("Failed to validate '{}' due to '{}'",
-                                            ident, e)));
-            }
+    opts.into_spec(&mut spec);
```

Comment:
Digging into this `into_spec` function / trait... it's a little weird.

One possibly interesting and more idiomatic refactoring might be to implement `From<SvcLoad> for ServiceSpec`, and then construct a way to "layer" two `ServiceSpec`s together.

That might also allow you to avoid a `mut` variable, which is often nice.

That's not a blocker or anything, but could be interesting to explore for a bit.